completions/smartctl

```diff
@@ -38,13 +72,42 @@ _comp_cmd_smartctl__drivedb()
     _comp_compgen_filedir h && [[ $prefix ]] &&
         _comp_compgen -Rv COMPREPLY -- -P "$prefix" -W '"${COMPREPLY[@]}"'
 }
+_comp_cmd_smartctl__vendorattribute()
+{
+    local fmt
+    case $cur in
+        [1-9N],* | [1-9][0-9],* | 1[0-9][0-9],* | 2[0-4][0-9],* | 25[0-5],*)
+            # TODO: add 'raw16(raw16)', 'raw16(avg16)' and 'raw24(raw8)'.
+            # Excluded for now because 'compopt -o fullquote' (Bash >= 5.3)
+            # or similar bash_completion functionality would be required.
+            fmt='raw{8,16,48,56,64},hex{48,56,64},raw24/raw{24,32}'
+            fmt+=',msec24hour32,{sec,min,halfmin}2hour,temp10x,tempminmax'
+            _comp_compgen -P "${cur%%,*}," -- -W '{'"$fmt"'}{,\,,:}'
```

Comment:
Showing three variations for all of the possible values (16x3 or 19x3) may be too much. It would be tedious to select one in `menu-complete`. What do you think about the idea of showing the three variations only when the candidate is unique?
```suggestion
            _comp_compgen -P "${cur%%,*}," -- -W '{'"$fmt"'}'
            if ((${#COMPREPLY[@]} == 1)); then
                _comp_compgen -R -- -W '"$COMPREPLY"{,\,,:}'
            fi
```

If we adopt this, `'{'"$fmt"'}'` should be expanded in the actual code.

Reply:
Makes plenty of sense && done